test/dt_arith.c

```diff
@@ -3167,864 +3838,629 @@ test_conv_flt_1_hw_conv_from_ldouble(void *hw_dst, unsigned char *src_buf, size_
 }
 #endif
 
+#ifdef H5_HAVE_COMPLEX_NUMBERS
 /*-------------------------------------------------------------------------
- * Function:    test_conv_flt_1
+ * Function:    test_conv_flt_1_hw_conv_from_fcomplex
  *
- * Purpose:    Test conversion of floating point values from SRC to
- *             DST.  These types should be H5T_NATIVE_FLOAT,
- *             H5T_NATIVE_DOUBLE, H5T_NATIVE_LDOUBLE or H5T_NATIVE_FLOAT16,
- *             if available.
+ * Purpose:     Helper function for test_conv_flt_1 to perform conversion
+ *              from float _Complex / _Fcomplex to another type by casting.
+ *              Also checks for overflow and underflow when the destination
+ *              type is a type with a smaller width than float.
  *
- * Return:    Success:    0
- *            Failure:    number of errors
+ * Return:      enum conv_func_ret_t value
  *
  *-------------------------------------------------------------------------
  */
-static int
-test_conv_flt_1(const char *name, int run_test, hid_t src, hid_t dst)
+static conv_func_ret_t
+test_conv_flt_1_hw_conv_from_fcomplex(void *hw_dst, unsigned char *src_buf, size_t idx, dtype_t dst_type)
```

Comment:
Another part of the diff that's not super useful. The logic in `test_conv_flt_1` was getting a bit lengthy due to all the different types supported. Similar to changes in the `_Float16` PR, I extracted out the logic that deals with casting a specific type to another type into helper functions starting here. `test_conv_flt_1` moved down below all these helper functions.